extensions/SiteMapIndex/Extension.pm

```diff
@@ -127,16 +124,6 @@ EOT
     };
 }
 
-sub _fix_robots_txt {
-    my ($self) = @_;
-    my $cgi_path = bz_locations()->{'cgi_path'};
-    my $robots_file = "$cgi_path/robots.txt";
-    my $current_fh = new IO::File("$cgi_path/robots.txt", 'r');
-    if (!$current_fh) {
-        warn "$robots_file: $!";
-        return;
-    }
-
     my $current_contents;
```

Comment:
delete the rest of the lines until _PACKAGE__->NAME; as well (_PACKAGE__->NAME should be the last line)

Before _PACKAGE__->NAME; add the following code (which is the before_robots_txt hook, which puts a single template variable, SITEMAP_URL)

```perl
sub before_robots_txt {
    my ($self, $args) = @_;
    $args->{vars}{SITEMAP_URL} = correct_urlbase() . SITEMAP_URL;
}
```